### PR TITLE
[ALLUXIO-3199]add try lock block method to BlockLockManager

### PR DESCRIPTION
**Pull request by Mitisky**
https://alluxio.atlassian.net/browse/ALLUXIO-3199

### CODE REVIEW

**Comment by Mitisky** [comment]
@apc999 PTAL.thanks

**Comment by Mitisky** [comment]
jenkins, test this please

**Comment by Mitisky** [comment]
jenkins, test this please

**Comment by Mitisky** [comment]
jenkins, test this please

**Comment by Mitisky** [comment]
@apc999 PTAL.Thanks

File: core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java [hunk 1]
```diff
@@ -802,16 +804,21 @@ private BlockMetadataManagerView getUpdatedView() {
...
    */
   private MoveBlockResult moveBlockInternal(long sessionId, long blockId,
-      BlockStoreLocation oldLocation, BlockStoreLocation newLocation)
-          throws BlockDoesNotExistException, BlockAlreadyExistsException,
-          InvalidWorkerStateException, IOException {
-    long lockId = mLockManager.lockBlock(sessionId, blockId, BlockLockType.WRITE);
+      BlockStoreLocation oldLocation, BlockStoreLocation newLocation, boolean wait)
+      throws BlockDoesNotExistException, BlockAlreadyExistsException,
+      InvalidWorkerStateException, IOException {
+    long lockId = wait ? mLockManager.lockBlock(sessionId, blockId, BlockLockType.WRITE) :
```
Comment (yupeng9) [inline core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java]: I feel the wait and retry logic should be handled by the client but not the server. @apc999 what do you think?

**Comment by Reidddddd** [comment]
Thanks for ping me here @Mitisky 
But i doubt it is a good solution, not only moveBlock but also other write behavior can get blocked.
And the root cause why is described in https://alluxio.atlassian.net/browse/ALLUXIO-3249
FYI @apc999

**Comment by Mitisky** [comment]
Thanks for your suggestion @Reidddddd 
Indeed your description is the root reason. My solution is to avoid competition between write operations.Do you have a good idea about this now?

**Comment by Mitisky** [comment]
@apc999 I encountered another situation the worker got stuck, which form a dead lock.Almost all thread is waiting to promote a block which run out of the thread pool.It will block my task forever and the locked block will never be released.I add a waiting time parameter in the promote method too.It can solve my problem,but I think this part code logic should be reviewed.

**Pull request closed without merging**